modules/nf-core/fast2q/tests/main.nf.test

```diff
@@ -0,0 +1,103 @@
+        when {
+            
+            params {
+                module_args = '--mo EC'
+            }
+
+            process {
+                """
+                input[0] = [
+                    [ id:'test1', single_end:true ], // meta map
+                    file(params.test_data_base + '/data/genomics/mus_musculus/mageck/ERR376998.small.fastq.gz', checkIfExists: true) // FASTQ file
+                ]
+                input[1] = [
+                    [ id:'test2', single_end:true ], // meta map for second input
+                    []
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.count_matrix).exists() } // Check processed count table exists
+                { assert path(process.out.versions).readLines().join("\n").contains("2FAST2Q version") } // Verify version output
+            )
+        }
+
+    }
+
+    test("2FAST2Q self-test") {
+
+        when {
+
+            params {
+            module_args = '-t'
+                }
+
+            process {
+                """
+                input[0] = [
+                    [ id:'test1', single_end:true ], // meta map
+                    file(params.test_data_base + '/data/genomics/mus_musculus/mageck/ERR376998.small.fastq.gz', checkIfExists: true) // FASTQ file
+                ]
+                input[1] = [
+                    [ id:'test2', single_end:true ], // meta map for second input
+                    []
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.count_matrix).exists() } // Check processed count table exists
+                { assert path(process.out.versions).readLines().join("\n").contains("2FAST2Q version") } // Verify version output
+            )
+        }
+
+    }
+
+    test("FASTQ file with optional library") {
+
+        when {
+
+            process {
+                """
+                input[0] = [
+                    [ id:'test1', single_end:true ], // meta map
+                    file(params.test_data_base + '/data/genomics/mus_musculus/mageck/ERR376998.small.fastq.gz', checkIfExists: true) // FASTQ file
+                ]
+                input[1] = [
+                    [ id:'test2', single_end:true ], // meta map for second input
+                    file(params.test_data_base + '/data/genomics/mus_musculus/mageck/yusa_library.csv', checkIfExists: true) // library file
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.count_matrix).exists() } // Check processed count table exists
+                { assert path(process.out.versions).readLines().join("\n").contains("2FAST2Q version") } // Verify version output
```

Comment:
The version should just contain the tool name, see [the guidelines](https://nf-co.re/docs/guidelines/components/modules#emission-of-versions).
You can capture the whole thing inside a snapshot using:
`path(process.out.versions[0]).yaml`

I would try for the assertion here (both checking the count_matrix and the versions):
{ assert snapshot(
            process.out,
            path(process.out.versions[0]).yaml
).match() },
(with correct indentation)